platform-sdk/swirlds-config-processor/src/main/java/com/swirlds/config/processor/ConfigDataAnnotationProcessor.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.config.processor;
+
+import static com.swirlds.config.processor.ConfigProcessorConstants.CONSTANTS_CLASS_SUFFIX;
+
+import com.google.auto.service.AutoService;
+import com.swirlds.config.processor.antlr.AntlrConfigRecordParser;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Processor;
+import javax.annotation.processing.RoundEnvironment;
+import javax.annotation.processing.SupportedAnnotationTypes;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.TypeElement;
+import javax.tools.Diagnostic.Kind;
+import javax.tools.FileObject;
+import javax.tools.JavaFileObject;
+import javax.tools.StandardLocation;
+
+/**
+ * An annotation processor that creates documentation and constants for config data records.
+ */
+@SupportedAnnotationTypes(ConfigProcessorConstants.CONFIG_DATA_ANNOTATION)
+@SupportedSourceVersion(SourceVersion.RELEASE_17)
+@AutoService(Processor.class)
+public final class ConfigDataAnnotationProcessor extends AbstractProcessor {
+
+    @Override
+    public boolean process(
+            final @NonNull Set<? extends TypeElement> annotations, final @NonNull RoundEnvironment roundEnv) {
+        Objects.requireNonNull(roundEnv, "annotations must not be null");
+        Objects.requireNonNull(roundEnv, "roundEnv must not be null");
+
+        if (annotations.isEmpty()) {
+            return false;
+        }
+        final String executionPath = System.getProperty("user.dir");
```

Comment:
Do you expect that the repo is cloned in the home directory?

Reply:
System.getProperty("user.dir") fetches the directory or path of the workspace for the current project